src/palimpzest/execution/plan_executors/single_threaded_execution_engine.py

```diff
@@ -0,0 +1,299 @@
+from palimpzest.corelib.schemas import SourceRecord
+from palimpzest.dataclasses import OperatorStats, PlanStats
+from palimpzest.elements import DataRecord
+from palimpzest.execution import ExecutionEngine
+from palimpzest.operators import AggregateOp, DataSourcePhysicalOp, LimitScanOp, MarshalAndScanDataOp
+from palimpzest.operators.filter import FilterOp
+from palimpzest.optimizer import PhysicalPlan
+
+from palimpzest.dataclasses import OperatorStats, PlanStats
+
+from typing import Optional, Union
+
+import time
+
+
+class SequentialSingleThreadPlanExecutor(ExecutionEngine):
+    """
+    This class implements the abstract execute_plan() method from the ExecutionEngine.
+    This class still needs to be sub-classed by another Execution class which implements
+    the higher-level execute() method.
+    """
+
+    def execute_plan(self, plan: PhysicalPlan,
+                     num_samples: Union[int, float] = float("inf"),
+                     max_workers: Optional[int] = None):
+        """Initialize the stats and the execute the plan."""
+        if self.verbose:
+            print("----------------------")
+            print(f"PLAN[{plan.plan_id}] (n={num_samples}):")
+            plan.printPlan()
```

Comment:
Is there any difference between plan.printPlan() and print(plan) ? If so, maybe we should unify

Reply:
`printPlan()` has slightly nicer syntax (the `inputSchema --> Operator --> outputSchema`) syntax; so I'll update `__str__` to be the body of `printPlan()`.